depinject/preference.go

```diff
@@ -0,0 +1,37 @@
+package depinject
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Preference defines a type binding preference to bind Interface to type Implementation when being provided as a
+// dependency to the module with ModuleName.  If ModuleName is empty then the type binding is applied globally,
+// not module-scoped.
+type Preference struct {
```

Comment:
is this exported because of the error?

Reply:
Yes I had hoped to construct a detailed error message from this struct in appConfig I think

Reply:
Seems to work OK un-exported though